iavl/iavl_tree.go

```diff
@@ -158,6 +167,14 @@ func (t *IAVLTree) GetByIndex(index int) (key []byte, value []byte) {
 	return t.root.getByIndex(t, index)
 }
 
+func (t *IAVLTree) GetWithProof(key []byte) ([]byte, *KeyExistsProof, *KeyNotExistsProof, error) {
+	value, proof := t.ConstructKeyExistsProof(key)
+	if proof == nil {
+		return nil, nil, nil, nil
```

Comment:
Wait, does this ever return error?  Either have an error case, or remove it from the signature please.